src/attribution/onCLS.ts

```diff
@@ -33,26 +32,31 @@ const getLargestLayoutShiftSource = (sources: LayoutShiftAttribution[]) => {
   return sources.find((s) => s.node && s.node.nodeType === 1) || sources[0];
 };
 
-const attributeCLS = (metric: CLSMetric): void => {
+const attributeCLS = (metric: CLSMetric): CLSMetricWithAttribution => {
+  // Use an empty object if no other attribution has been set.
+  let attribution: CLSAttribution = {};
+
   if (metric.entries.length) {
     const largestEntry = getLargestLayoutShiftEntry(metric.entries);
     if (largestEntry && largestEntry.sources && largestEntry.sources.length) {
       const largestSource = getLargestLayoutShiftSource(largestEntry.sources);
       if (largestSource) {
-        (metric as CLSMetricWithAttribution).attribution = {
+        attribution = {
           largestShiftTarget: getSelector(largestSource.node),
           largestShiftTime: largestEntry.startTime,
           largestShiftValue: largestEntry.value,
           largestShiftSource: largestSource,
           largestShiftEntry: largestEntry,
           loadState: getLoadState(largestEntry.startTime),
         };
-        return;
       }
     }
   }
-  // Set an empty object if no other attribution has been set.
-  (metric as CLSMetricWithAttribution).attribution = {};
+
+  // Cast to attribution metric so it can be populated.
+  const metricWithAttribution = metric as CLSMetricWithAttribution;
```

Comment:
What about doing something like this to avoid having to typecast?

```ts
const metricWithAttribution: CLSMetricWithAttribution = Object.assign(
  metric,
  {attribution},
);

return metricWithAttribution;
```

Reply:
Oh, did I misunderstand the need to maintain object identity for the metrics? e.g. so Set will work with multiple callbacks for a metric

Reply:
This is maintaining object identity, no?

The main reason I was suggesting this is to appease the TypeScript gods and not use type assertions. I don't think it's functionally better than what you originally had.

Reply:
Ugh, yeah, I need to read less quickly (I guess I was anchored on the `Object.assign({}, ...)` spread replacement). Annoying we need to run that line but great to get rid of the type assertions altogether.

Reply:
Updated!